File: backend/app/schemas/workspace.py
from pydantic import BaseModel

from app.schemas.project import ProjectResponse
from app.schemas.paper import PaperResponse
from app.schemas.draft import DraftResponse


class WorkspaceStats(BaseModel):

    papers: int

    drafts: int

    reviews: int


class WorkspaceResponse(BaseModel):

    project: ProjectResponse

    papers: list[PaperResponse]

    drafts: list[DraftResponse]

    stats: WorkspaceStats

File: Client/paper-mind/src/types/workspace.ts
import type { Draft } from "./draft";
import type { Paper } from "./paper";
import type { Project } from "./project";

export interface WorkspaceStats {
  papers: number;
  drafts: number;
  reviews: number;
}

export interface Workspace {
  project: Project;
  papers: Paper[];
  drafts: Draft[];
  stats: WorkspaceStats;
}